trunner/test_runner.py

```diff
@@ -20,6 +25,69 @@ def resolve_project_path():
     return project_dir
 
 
+def init_logdir(logdir):
+    """Inits log directories if it's needed."""
+    if not logdir:
+        return
+
+    # clear the whole log directory before the next campaign
+    if os.path.exists(logdir):
+        shutil.rmtree(logdir)
+
+    # test campaign directory will always be needed
+    os.makedirs(f"{logdir}/test_campaign")
+
+
+def set_logfiles(dut: Dut):
+    """Sets dut logfiles associated with the pexpect process if needed."""
+
+    if not trunner.ctx.logdir and not is_github_actions():
```

Comment:
Please pass logdir as an argument to the `set_logfiles` function. Let's try to be consistent in all function headers.